File: problems/3163-subarrays-distinct-element-sum-of-squares-i/solution.ex
defmodule Solution do
  def sum_counts(nums) do
    Enum.reduce(1..length(nums), 0, fn i, acc ->
      acc + Enum.reduce(0..(length(nums)-i), 0, fn j, sub_acc ->
        sub_acc + Enum.count(Enum.uniq(Enum.slice(nums, j, i))) ** 2
      end)
    end)
  end
end

File: problems/3163-subarrays-distinct-element-sum-of-squares-i/solution.py
class Solution(object):
    def sumCounts(self, nums):
        ans, l = 0, len(nums)
        
        for i in range(1, l+1):
            for j in range(0, l-i+1):
                ans += len(set(nums[j:i+j])) ** 2
        
        return ans  
